shard/committee/assignment.go

```diff
@@ -159,20 +159,22 @@ func eposStakedCommittee(
 		return shardState, nil
 	}
 
+	maxBLSKey := stakedSlotsCount / 3
+
 	// TODO benchmark difference if went with data structure that sorts on insert
 	for i := range candidates {
 		validator, err := stakerReader.ReadValidatorInformation(candidates[i])
 		if err != nil {
 			return nil, err
 		}
-		if !effective.IsEligibleForEPOSAuction(validator) {
+
+		if validator.EPOSStatus != effective.Active {
 			continue
 		}
-		if err := validator.SanityCheck(); err != nil {
+		if err := validator.SanityCheck(maxBLSKey); err != nil {
```

Comment:
This input of MaxBLSKey and everywhere is too verbose. Why not directly add the check inside SanityCheck using "stakedSlots :=
		(instance.NumNodesPerShard() - instance.NumHarmonyOperatedNodesPerShard()) *
			int(instance.NumShards())" ? 

Then we don't need all these wiring.

Reply:
I thought about that too but its very awkward to grab a handle on the instance since I don't have the epoch , so I must provide some function parameter

Reply:
ok